### PR TITLE
Image endpoints

### PR DESCRIPTION
**Pull request by AlDu2407**
`GET` endpoint:
- Added new handler for fetching previously uploaded images from Urchin.
- Added structs to allow handling of responses with compiler support.
- Added required functions for the database to fetch images.

`DELETE` endpoint:
- Added new handler for deleting previously uploaded images from Urchin.
- Added required functions for the database to delete images.

General update:
- Added structs to allow handling of responses with compiler support.
- Restructured API to used path variables for the delete endpoints.

### CODE REVIEW

File: tests/admin_app_tests/endpoint_tests/images_test.go [hunk 1]
```diff
@@ -181,7 +183,225 @@ func TestPostImageFailedToCreateDatabaseEntry(t *testing.T) {
...
+		}
 
+		err = png.Encode(part, img)
+		if err != nil {
+			t.Error(err)
+		}
+	}()
+
+	// TODO: We have to create the image first. Maybe there's a better way to do this?
+	req, _ := http.NewRequest("POST", "/images", pr)
```
Comment (matheusgomes28) [inline tests/admin_app_tests/endpoint_tests/images_test.go:230]: There is another way: we could mock the access to the image files. I'm thinking of doing something like a `MediaVault` struct, then that could have `AddImage(...)`, `GetImage(...)` etc... But this is fine for this PR. I'll create some tasks for this

**Review by matheusgomes28 — COMMENTED** [review]
This is really good Al, thank you for the feature! I will suggest: let's move the current `GET /image` endpoint to the main `app` (non-admin), and for the admin we can simply return the image metadata (i.e. the `Image` struct filled with the DB data).

Remember: the admin-app does not need to be running for the app to work, and the file serving should definitely come from the main app 😄

**Comment by AlDu2407** [comment]
I added the required functionality to the app. Also, refactored the `GET` image endpoint in the admin app. 
Improved the handling of the NavBar.

Hope this is fine for you.

File: tests/admin_app_tests/endpoint_tests/images_test.go (hunk 1 above)
Comment (AlDu2407) [reply tests/admin_app_tests/endpoint_tests/images_test.go:230]: Should be moved to a separate task. For now, I wouldn't want to increase the size of this PR further more.

**Review by matheusgomes28 — COMMENTED** [review]
I like the web pages for the image & image gallery to showcase the added images. 

I'll take another look and test it all tomorrow morning, for some reason I'm having issues connecting to the docker composed mariadb so I can do a few e2e tests on this.

Sorry to delay this Al, but I just wanna do a quick usability test before we merge!

File: tests/app_tests/endpoint_tests/posts_test.go [hunk 2]
```diff
@@ -34,7 +34,7 @@ func TestPostSuccess(t *testing.T) {
 
 	r := app.SetupRoutes(app_settings, database_mock)
 	w := httptest.NewRecorder()
-	req, _ := http.NewRequest("GET", "/post/0", nil)
+	req, _ := http.NewRequest("GET", "/post/1", nil)
```
[the remark this answers is not in the document]
Comment (AlDu2407) [reply tests/app_tests/endpoint_tests/posts_test.go:37]: I have no clue, but for some reason `0` as an ID makes the test fail.

[the remark this answers is not in the document]
Comment (AlDu2407) [reply tests/app_tests/endpoint_tests/posts_test.go:37]: I tried to take a look into it and I assume that `0` as an ID is interpreted as empty/nil and therefore the `required` part of the URI binding is not fulfilled. Might be a quirk from the Gin Framework and using their `ShouldBindUri(...)` function.

File: tests/admin_app_tests/endpoint_tests/images_test.go (hunk 1 above)
Comment (matheusgomes28) [reply tests/admin_app_tests/endpoint_tests/images_test.go:230]: Yes

File: tests/app_tests/endpoint_tests/posts_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (matheusgomes28) [reply tests/app_tests/endpoint_tests/posts_test.go:37]: Weird, I'll investigate this. Let's slide this in for now.

**Review by matheusgomes28 — APPROVED** [review]
I was finally able to test this properly. Turns out I was running `docker compose up mariadb` which is not the same as `docker compose -f mariadb.yml up -d` ...

Looks fine, let's merge this in then we can perhaps weed out some of the minor stuff later. The things I particularly don't want to keep in the long run are:

- Absolute sizes for the margins and other CSS stuff - doesn't play well with different resolutions.
- Fixed links being returned in the `GetUrchinLinks` - this would be good to get programatically from the user somehow (let them provide nav links). I'll create a feature task for this.
- Weird failure when you have the post mock id set to `0`, it seemed to be passing in the previous pipelines so I gotta figure out what's going on.
- Generally remove the handwritten CSS - I dabbled with `tailwind` before but unfortunately I'm not a very good designer, so let's hope I can find someone to make this all look nicer !

**Comment by matheusgomes28** [comment]
I need you to rebase the new main changes on this branch so I can merge!

**Pull request merged**